openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/util/ExpiringSpeedStorageSpeedCalculator.java

```diff
@@ -0,0 +1,34 @@
+package org.heigit.ors.routing.graphhopper.extensions.util;
+
+import com.graphhopper.routing.EdgeKeys;
+import com.graphhopper.routing.profiles.DecimalEncodedValue;
+import com.graphhopper.routing.util.AbstractAdjustedSpeedCalculator;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.routing.util.SpeedCalculator;
+import com.graphhopper.util.EdgeIteratorState;
+import org.heigit.ors.routing.graphhopper.extensions.storages.ExpiringSpeedStorage;
+
+/**
+ * Simple SpeedCalculator based on a SpeedStorage.
+ * Default speed of Byte.MIN_VALUE results in original graph speed
+ */
+public class ExpiringSpeedStorageSpeedCalculator extends AbstractAdjustedSpeedCalculator {
+    private ExpiringSpeedStorage expiringSpeedStorage;
+
+    public ExpiringSpeedStorageSpeedCalculator(SpeedCalculator superSpeedCalculator, ExpiringSpeedStorage expiringSpeedStorage) {
```

Comment:
How about simplifying the class name to be `ExpiringSpeedCalculator`?

On the general note, I was thinking about the whole concept of speed values valid only within a limited time window. Supposed the storage contains entries from 10min ago, while the validity is set to 15min. Then in principle the speed values are expected to be valid only for another 5min, right? How about routes which take longer than that. For example, for a 10min route already half-way into it the speed values would be considered as outdated. Just wondering whether this point has been ever raised and discussed with the client?

Reply:
Renamed.

Just because the validity of an entry is 15m doesn't mean that in 15m driving time the route will not have any traffic anymore. The idea is that speeds will be updated every now and then while they persist. Implementing it in the way you suggest would mean to never use traffic that is further than 15m away, which would be wrong. I see the point, but it cannot be solved in this way or any other easy way, as you need a traffic prediction for that. The current implementation has a "prediction", which is the current traffic. That is the best prediction we can offer.
The idea of expiration was a customer request.

Reply:
Got it, thanks for clarifying!